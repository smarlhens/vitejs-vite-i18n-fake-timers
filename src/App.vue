<script setup lang="ts">
import HelloWorld from './components/hello-world.vue';
import HelloWorldI18n from './components/hello-world-i18n.vue';
</script>

<template>
  <HelloWorld />
  <HelloWorldI18n />
</template>
