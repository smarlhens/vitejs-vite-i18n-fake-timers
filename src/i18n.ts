import { createI18n } from 'vue-i18n';

import common from '../locales/common.json';

export const i18n = createI18n({
    legacy: false,
    globalInjection: true,
    fallbackLocale: 'en',
    locale: 'en',
    messages: common,
});
