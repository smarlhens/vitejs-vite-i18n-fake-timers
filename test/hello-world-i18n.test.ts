import type { RenderResult } from '@testing-library/vue';
import { render } from '@testing-library/vue';
import { afterEach, beforeEach, describe, expect, it } from 'vitest';

import HelloWorldI18n from '../src/components/hello-world-i18n.vue';
import { i18n } from '../src/i18n';

describe('HelloWorldI18n', () => {
    let wrapper: RenderResult;

    afterEach(() => {
        wrapper.unmount();
    });

    beforeEach(() => {
        const mountOptions = {
            global: {
                plugins: [i18n],
            }
        }

        wrapper = render(HelloWorldI18n, mountOptions);
    });

    it('should render', () => {
        const { html, getByText } = wrapper;
        getByText('HelloWorld');
        expect(html()).toMatchSnapshot();
    });
});
