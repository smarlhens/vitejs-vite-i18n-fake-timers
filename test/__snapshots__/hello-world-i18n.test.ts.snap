// Vitest Snapshot v1

exports[`HelloWorldI18n > should render 1`] = `"<h1>HelloWorld</h1>"`;
